packages/legend-query/src/components/QueryBuilderPostFilterPanel.tsx

```diff
@@ -0,0 +1,1069 @@
+);
+
+export const QueryBuilderPostFilterPanel = observer(
+  (props: { queryBuilderState: QueryBuilderState }) => {
+    const { queryBuilderState } = props;
+    const applicationStore = useApplicationStore();
+    const postFilterState = queryBuilderState.postFilterState;
+    const rootNode = postFilterState.getRootNode();
+    // actions
+    const collapseTree = (): void => {
+      postFilterState.setSelectedNode(undefined);
+      postFilterState.collapseTree();
+    };
+    const expandTree = (): void => {
+      postFilterState.setSelectedNode(undefined);
+      postFilterState.expandTree();
+    };
+    const pruneTree = (): void => {
+      postFilterState.suppressClickawayEventListener();
+      postFilterState.pruneTree();
+    };
+    const simplifyTree = (): void => {
+      postFilterState.suppressClickawayEventListener();
+      postFilterState.simplifyTree();
+    };
+    const createCondition = (): void => {
+      postFilterState.suppressClickawayEventListener();
+      postFilterState.addNodeFromNode(
+        new QueryBuilderPostFilterTreeBlankConditionNodeData(undefined),
+        postFilterState.selectedNode,
+      );
+    };
+    const allowGroupCreation =
+      postFilterState.isEmpty || // either the tree is empty
+      (postFilterState.selectedNode && // or a node is currently selected which is...
+        (postFilterState.selectedNode !== rootNode || // either not a root node
+          rootNode instanceof QueryBuilderPostFilterTreeGroupNodeData)); // or if it is the root note, it has to be a group node
+
+    const createGroupCondition = (): void => {
+      postFilterState.suppressClickawayEventListener();
+      if (allowGroupCreation) {
+        postFilterState.addGroupConditionNodeFromNode(
+          postFilterState.selectedNode,
+        );
+      }
+    };
+    const newGroupWithCondition = (): void => {
+      postFilterState.suppressClickawayEventListener();
+      if (
+        postFilterState.selectedNode instanceof
+        QueryBuilderPostFilterTreeConditionNodeData
+      ) {
+        postFilterState.newGroupWithConditionFromNode(
+          undefined,
+          postFilterState.selectedNode,
+        );
+      }
+    };
+    // Drag and Drop
+    const handleDrop = useCallback(
+      async (item: QueryBuilderPostFilterDropTarget): Promise<void> => {
+        let postFilterConditionState: PostFilterConditionState;
+        try {
+          const columnState = (item as QueryBuilderProjectionColumnDragSource)
+            .columnState;
+          const aggregateColumnState =
+            queryBuilderState.fetchStructureState.projectionState.aggregationState.columns.find(
+              (column) => column.projectionColumnState === columnState,
+            );
+          if (
+            !aggregateColumnState &&
+            columnState instanceof QueryBuilderDerivationProjectionColumnState
+          ) {
+            await flowResult(columnState.fetchDerivationLambdaReturnType());
+          }
+
```

Comment:
Should we `return` here if the return-type is never sucessfully fetched?

Reply:
yeah. changes the function to throw if there is an error in failing. The error is handled by consumers